File: app/src/main/java/com/wilderpereira/lmgtfygen/domain/repository/UrlShortenerApi.java
package com.wilderpereira.lmgtfygen.domain.repository;

import com.wilderpereira.lmgtfygen.domain.entity.ShortenerResponse;

import retrofit2.http.Field;
import retrofit2.http.POST;
import rx.Observable;

/**
 * Created by Wilder on 30/01/17.
 */

public interface GoogleShortenerApi {
    @POST("url")
    Observable<ShortenerResponse> shortenUrl(@Field("longUrl")  String longUrl, @Field("key") String key);
}

File: app/src/main/java/com/wilderpereira/lmgtfygen/domain/repository/UrlShortenerApi.kt
package com.wilderpereira.lmgtfygen.domain.repository

import com.wilderpereira.lmgtfygen.domain.entity.SearchUrl
import com.wilderpereira.lmgtfygen.domain.entity.ShortenerBody

import retrofit2.http.Body
import retrofit2.http.POST
import retrofit2.http.Query
import rx.Observable

/**
 * Created by Wilder on 30/01/17.
 */

interface UrlShortenerApi {
    @POST("url")
    fun shortenUrl(@Query("key") key: String, @Body body: ShortenerBody): Observable<SearchUrl>
}
